Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -333,6 +346,33 @@ private Branch(Context context) {
 		linkCache_ = new HashMap<BranchLinkData, String>();
 	}
 
+
+	/**
+	 * <p>Singleton method to return the pre-initialised object of the type {@link Branch}.
+	 * Make sure your app is  instantiating {@link BranchApp} before calling this method</p>
+	 *
+	 * @return An initialised singleton {@link Branch} object
+	 *
+	 * @throws BranchException Exception</br>
+	 *          1)If your {@link Application}  is not instance of {@link BranchApp} </br>
+	 *          2)If the minimum API level is below 14
+	 */
+	public static Branch getInstance() throws BranchException {
```

Comment:
I'm not a big fan of having to make the developer wrap all our method calls in try/catch personally. It always annoyed me that I had to do it for JSON. Can we just silently fail and throw an error log? Thoughts?